app/services/store_sp_metadata_in_session.rb

```diff
@@ -24,20 +32,76 @@ def sp_request
     @sp_request ||= ServiceProviderRequestProxy.from_uuid(request_id)
   end
 
+  def ial_value
+    if parsed_vot.ialmax?
+      0
+    elsif parsed_vot.identity_proofing?
+      2
+    else
+      1
+    end
+  end
+
+  def ial2_value
+    parsed_vot.identity_proofing?
+  end
+
+  def ialmax_value
+    parsed_vot.ialmax?
+  end
+
+  def aal_level_requested_value
+    if parsed_vot.phishing_resistant?
+      3
+    elsif parsed_vot.aal2?
+      2
+    else
+      1
+    end
+  end
+
+  def piv_cac_requested_value
+    parsed_vot.hspd12?
+  end
+
+  def phishing_resistant_value
+    parsed_vot.phishing_resistant?
+  end
+
+  def biometric_comparison_required_value
+    parsed_vot.biometric_comparison?
+  end
+
   def update_session
-    session[:sp] = {
-      issuer: sp_request.issuer,
-      ial: ial_context.ial,
-      ial2: ial_context.ial2_requested?,
-      ialmax: ial_context.ialmax_requested?,
-      aal_level_requested: aal_requested,
-      piv_cac_requested: hspd12_requested,
-      phishing_resistant_requested: phishing_resistant_requested,
-      request_url: sp_request.url,
-      request_id: sp_request.uuid,
-      requested_attributes: sp_request.requested_attributes,
-      biometric_comparison_required: sp_request.biometric_comparison_required,
-    }
+    if IdentityConfig.store.use_vot_in_sp_requests
```

Comment:
I don't think we need this conditional.

The VOT Parser can handle ACR values. We should be able to pass both the ACR and VOT value into the parser and have it work.

I think the feature flag should prevent us from consuming and storing the VTR param rather than from using it here.

Reply:
I just integrated the feature flag into the form where the VTR is consumed here: https://github.com/18F/identity-idp/pull/10004/commits/1cb5ed4a0c6abf9ec8d7e122aac662fb755ec290